examples/help_categories.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
```

Comment:
The example is nice and helps explain to end users how to use it.

A couple unit tests covering the new features added would be extremely beneficial from a maintenance perspective.